### PR TITLE
Refactor TMW Stats (API + Fix Update Issues in Frontend)

### PR DESCRIPTION
**Pull request by jpennors**

### CODE REVIEW

File: app/StatTag.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+     * 
+     */
+    public static function launchStatTagJob($tag_id, $action)
+    {
+        StatTagJob::dispatch($tag_id, $action);
+    }
+
+
+    /**
+     * Method to save specifique action on Resource
```
Comment (lancelotimb) [inline app/StatTag.php]:
```suggestion
     * Method to save specific action on Resource
```

File: react/app-admin/components/Statistics/SearchTagsChart.tsx [hunk 2]
```diff
@@ -3,164 +3,158 @@ import { Form, Select } from 'semantic-ui-react';
...
+    const selectDefaultData = (): number[] => {
+        return searchTagsStats.filter(t => t.is_parent).map(t => t.count);
+    };
 
-    const updateChart = (labels: string[], data: number[]) : void => {
+    const updateChart = (labels: string[], data: number[]): void => {
         while (chart?.data.labels != null && chart.data.labels.length > 0) {
             chart.data.labels.pop();
         }
         labels.map(l => chart?.data.labels?.push(l));
```
Comment (lancelotimb) [inline react/app-admin/components/Statistics/SearchTagsChart.tsx]:
```suggestion
        labels.forEach(l => chart?.data.labels?.push(l));
```
Same effect but technically `forEach` is what you want here as you don't use the returned result

File: react/app-admin/components/Statistics/SearchTagsChart.tsx [hunk 3]
```diff
@@ -3,164 +3,158 @@ import { Form, Select } from 'semantic-ui-react';
...
+                text: 'Tags principaux',
+            },
         ];
-        searchTagsStats.map(t => options.push({
-            key: t.slug,
-            value: t.slug,
-            text: t.name,
-        }));
+        searchTagsStats.filter(t => t.is_parent)
+            .map(t =>
```
Comment (lancelotimb) [inline react/app-admin/components/Statistics/SearchTagsChart.tsx]: You can use the `convertToSelectOptions` in `utils/select-options` to convert tags to select options 🙂

**Review by lancelotimb — APPROVED** [review]

File: react/app-admin/components/Statistics/SearchTagsChart.tsx (hunk 3 above)
Comment (jpennors) [reply react/app-admin/components/Statistics/SearchTagsChart.tsx]: Done !

**Pull request merged**